packages/widgets/src/dns-hole/controls/component.tsx

```diff
@@ -0,0 +1,159 @@
+"use client";
+
+import { useState } from "react";
+import type { BoxProps } from "@mantine/core";
+import { ActionIcon, Badge, Box, Button, Card, Flex, Image, Tooltip, UnstyledButton } from "@mantine/core";
+import { useDisclosure } from "@mantine/hooks";
+import { IconClockPause, IconPlayerPlay, IconPlayerStop } from "@tabler/icons-react";
+
+import { clientApi } from "@homarr/api/client";
+import { integrationDefs } from "@homarr/definitions";
+import { useI18n } from "@homarr/translation/client";
+
+import type { WidgetComponentProps, WidgetProps } from "../../definition";
+import { NoIntegrationSelectedError } from "../../errors";
+import TimerModal from "./TimerModal";
+
+const dnsLightStatus = (currentStatus: string): "green" | "red" => {
+  if (currentStatus === "enabled") {
+    return "green";
+  }
+  return "red";
+};
+
+export default function DnsHoleControlsWidget({ options, integrationIds }: WidgetComponentProps<"dnsHoleControls">) {
+  const integrationId = integrationIds.at(0);
+
+  if (!integrationId) {
+    throw new NoIntegrationSelectedError();
+  }
+
+  const PiHoleIntegrationDef = integrationDefs.piHole;
+  const AdGuardHomeIntegrationDef = integrationDefs.adGuardHome;
+  const [data] = clientApi.widget.dnsHole.summary.useSuspenseQuery(
+    {
+      integrationIds,
+    },
+    {
+      refetchOnMount: false,
+      retry: false,
+    },
+  );
+
+  const t = useI18n();
+  const [status, setStatus] = useState<string>(data[0]?.summary.status);
```

Comment:
Currently, you only store one state. However, you want to support multiple instances.
For that, you need to reference multiple states using a key.
Since the integration ID is unique, we can use this as the key.
Example (replace ....... with actual code):
```ts
const [status, setStatus] = useState<{integrationId: string, enabled: boolean}[]>(................);
```